subgraphs/compound-forks/protocols/tectonic/src/mapping.ts

```diff
@@ -0,0 +1,380 @@
+}
+
+export function handleMint(event: Mint): void {
+  let minter = event.params.minter;
+  let mintAmount = event.params.mintAmount;
+  _handleMint(comptrollerAddr, minter, mintAmount, event);
+}
+
+export function handleRedeem(event: Redeem): void {
+  let redeemer = event.params.redeemer;
+  let redeemAmount = event.params.redeemAmount;
+  _handleRedeem(comptrollerAddr, redeemer, redeemAmount, event);
+}
+
+export function handleBorrow(event: BorrowEvent): void {
+  let borrower = event.params.borrower;
+  let borrowAmount = event.params.borrowAmount;
+  _handleBorrow(comptrollerAddr, borrower, borrowAmount, event);
+}
+
+export function handleRepayBorrow(event: RepayBorrow): void {
+  let payer = event.params.payer;
+  let repayAmount = event.params.repayAmount;
+  _handleRepayBorrow(comptrollerAddr, payer, repayAmount, event);
+}
+
+export function handleLiquidateBorrow(event: LiquidateBorrow): void {
+  let cTokenCollateral = event.params.cTokenCollateral;
+  let liquidator = event.params.liquidator;
+  let borrower = event.params.borrower;
+  let seizeTokens = event.params.seizeTokens;
+  let repayAmount = event.params.repayAmount;
+  _handleLiquidateBorrow(
+    comptrollerAddr,
+    cTokenCollateral,
+    liquidator,
+    borrower,
+    seizeTokens,
+    repayAmount,
+    event
+  );
+}
+
+// export function _handleAccrueInterest_Tectonic(event: ):
+
+export function handleAccrueInterest(event: AccrueInterest): void {
+  let marketAddress = event.address;
+  let cTokenContract = CToken.bind(marketAddress);
+  let protocol = getOrCreateProtocol();
+  let oracleContract = PriceOracle.bind(
+    Address.fromString(protocol._priceOracle)
+  );
+  let updateMarketData = new UpdateMarketData(
+    cTokenContract.try_totalSupply(),
+    cTokenContract.try_exchangeRateStored(),
+    cTokenContract.try_supplyRatePerBlock(),
+    cTokenContract.try_borrowRatePerBlock(),
+    oracleContract.try_getUnderlyingPrice(marketAddress),
+    SECONDS_PER_YEAR
+  );
+  let interestAccumulated = event.params.interestAccumulated;
+  let totalBorrows = event.params.totalBorrows;
+
+  let marketID = marketAddress.toHexString();
+  let market = Market.load(marketID);
+  if (!market) {
+    log.warning("[handleAccrueInterest] Market not found: {}", [marketID]);
+    return;
+  }
+  updateRewards(event, market);
+
+  _handleAccrueInterest(
+    updateMarketData,
+    comptrollerAddr,
+    interestAccumulated,
+    totalBorrows,
+    true,
+    event
+  );
+}
+
+function getOrCreateProtocol(): LendingProtocol {
+  let comptroller = Comptroller.bind(comptrollerAddr);
+  let protocolData = new ProtocolData(
+    comptrollerAddr,
+    "Tectonic",
+    "tectonic",
+    "1.3.0",
+    "1.0.0",
+    "1.0.0",
+    Network.CRONOS,
+    comptroller.try_liquidationIncentiveMantissa(),
+    comptroller.try_oracle()
+  );
+  return _getOrCreateProtocol(protocolData);
+}
+
+function updateRewards(event: ethereum.Event, market: Market): void {
```

Comment:
The naming should be cleaned up for this function

The subgraph isn't fully loaded as I am reviewing, but I am surprised there are no rewards at this point. I would do some more investigation here. Rewards and/or price might be returning 0 when not expected. 

Looking into it a little it is that the price is not being returned, when I am reviewing it should be getting the price from the market after block 1337194